ConcernsCaseWork/Service.TRAMS.Tests/Teams/TeamsServiceTests.cs

```diff
@@ -0,0 +1,150 @@
+using System.Threading.Tasks;
+
+namespace Service.TRAMS.Tests.Teams
+{
+	public class TeamsServiceTests
+	{
+		[Test]
+		public void Methods_GuardAgainstNullArgs()
+		{
+			var fixture = new AutoFixture.Fixture();
+			fixture.Customize(new AutoMoqCustomization());
+			var assertion = fixture.Create<GuardClauseAssertion>();
+			assertion.Verify(typeof(TeamsService).GetMethods());
+		}
+
+		[Test]
+		public void Constructors_GuardAgainstNullArgs()
+		{
+			var fixture = new AutoFixture.Fixture();
+			fixture.Customize(new AutoMoqCustomization());
+			var assertion = fixture.Create<GuardClauseAssertion>();
+			assertion.Verify(typeof(TeamsService).GetConstructors());
+		}
+
+		[Test]
+		public async Task GetTeam_Returns_Dto()
+		{
+			var expectedDto = new ConcernsCaseworkTeamDto("user.one", new[] { "user.two" });
+			var responseWrapper = new ApiWrapper<ConcernsCaseworkTeamDto>(expectedDto);
+			var configuration = new ConfigurationBuilder().ConfigurationUserSecretsBuilder().Build();
+			var tramsApiEndpoint = configuration["trams:api_endpoint"];
+			var httpClientFactory = new Mock<IHttpClientFactory>();
+			var mockMessageHandler = new Mock<HttpMessageHandler>();
+			mockMessageHandler.Protected()
+				.Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+				.ReturnsAsync(new HttpResponseMessage
+				{
+					StatusCode = HttpStatusCode.OK,
+					Content = new ByteArrayContent(JsonSerializer.SerializeToUtf8Bytes(responseWrapper))
+				});
+
+			var httpClient = new HttpClient(mockMessageHandler.Object);
+			httpClient.BaseAddress = new Uri(tramsApiEndpoint);
+			httpClientFactory.Setup(_ => _.CreateClient(It.IsAny<string>())).Returns(httpClient);
+
+			var sut = new TeamsService(httpClientFactory.Object, Mock.Of<ILogger<TeamsService>>());
+			var result = await sut.GetTeam("user.one");
+
+			Assert.AreEqual(expectedDto.OwnerId, result.OwnerId);
+			Assert.AreEqual(expectedDto.TeamMembers, result.TeamMembers);
+		}
+
+		[Test]
+		public async Task GetTeam_Allows_NoContent_Result()
+		{
+			var configuration = new ConfigurationBuilder().ConfigurationUserSecretsBuilder().Build();
+			var tramsApiEndpoint = configuration["trams:api_endpoint"];
+			var httpClientFactory = new Mock<IHttpClientFactory>();
+			var mockMessageHandler = new Mock<HttpMessageHandler>();
+			mockMessageHandler.Protected()
+				.Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+				.ReturnsAsync(new HttpResponseMessage
+				{
+					StatusCode = HttpStatusCode.NoContent
+				});
+
+			var httpClient = new HttpClient(mockMessageHandler.Object);
+			httpClient.BaseAddress = new Uri(tramsApiEndpoint);
+			httpClientFactory.Setup(_ => _.CreateClient(It.IsAny<string>())).Returns(httpClient);
+
+			var sut = new TeamsService(httpClientFactory.Object, Mock.Of<ILogger<TeamsService>>());
+			var result = await sut.GetTeam("user.one");
```

Comment:
This is more of a comment on the implementation than the test, but would we consider returning an empty collection rather than a null? V. minor.

Reply:
I figured I'd let the service return a null, and then the caller use an empty collection. That way you'd know from the service whether you got nothing or really got an empty one.